packages/eas-cli/src/commandUtils/context/DynamicProjectConfigContextField.ts

```diff
@@ -0,0 +1,31 @@
+import { ExpoConfig } from '@expo/config-types';
+
+import { ExpoConfigOptions, getExpoConfig } from '../../project/expoConfig';
+import ContextField, { ContextOptions } from './ContextField';
+import ProjectConfigContextField from './ProjectConfigContextField';
+import ProjectDirContextField from './ProjectDirContextField';
+
+export type DynamicConfigContextFn = (options?: ExpoConfigOptions) => Promise<{
+  projectId: string;
+  exp: ExpoConfig;
+  projectDir: string;
+}>;
+
+export class DynamicProjectConfigContextField extends ContextField<DynamicConfigContextFn> {
+  async getValueAsync({ nonInteractive }: ContextOptions): Promise<DynamicConfigContextFn> {
+    return async (options?: ExpoConfigOptions) => {
+      const projectDir = await ProjectDirContextField['findProjectDirAndVerifyProjectSetupAsync']();
```

Comment:
Yeah, that's a good point. I think what I wan't to avoid is people accidentally calling these from commands (with the exception of a few very specific commands that modify this context like `eas init`), and I couldn't think of a better way to do this than private methods and private method accesses.

Put another way, I can't think of another way to prevent these accesses in normal commands.

Reply:
I'll land this and follow-up the stack with an attempt to move these out to see how it looks.